File: codewars/6kyu/Counting Duplicates.js
function duplicateCount(input) {
    let lowerCaseInput = input.toLowerCase();

    let charCount = {};

    for (let char of lowerCaseInput) {
        if (charCount[char]) {
            charCount[char]++;
        } else {
            charCount[char] = 1;
        }
    }

    let duplicates = 0;
    for (let char in charCount) {
        if (charCount[char] > 1) {
            duplicates++;
        }
    }

    return duplicates;
}

console.log(duplicateCount('abcde')); // 0
console.log(duplicateCount('aabbcde')); // 2

File: codewars/6kyu/Counting Duplicates.py
from collections import Counter


def duplicate_count(text):
    counter = Counter([t.lower() for t in text])
    dup_list = [c for a, c in counter.items() if c > 1]
    return len(dup_list)


if __name__ == "__main__":
    print(duplicate_count("abcde"))  # 0
    print(duplicate_count("aabbcde"))  # 2
    print(duplicate_count("aabBcde"))  # 2
    print(duplicate_count("indivisibility"))  # 1
    print(duplicate_count("Indivisibilities"))  # 2
    print(duplicate_count("aA11"))  # 2
    print(duplicate_count("ABBA"))  # 2